### PR TITLE
refactor: modified names of resources to fit MS naming standard

### PR DESCRIPTION
**Pull request by anthonygauthier**
This PR renames the different resources according to [these standards](https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-naming).

### CODE REVIEW

**Review by christopherswarren — APPROVED** [review]
I can't find anything to complain about here. :) Nice work!

File: foundation/shared-services/core/managed-identity.tf
Comment (anthonygauthier) [reply foundation/shared-services/core/managed-identity.tf:35]: I thought about that, I believe it's a good idea personally, I just didn't go through with it. @djtjwillia What do you think?

Comment (jbankes) [reply foundation/shared-services/core/managed-identity.tf:35]: I think it would make sense to include it. Did we just include it because it wasn't listed? I'm not married to any particular idea but wanted to weigh in here.

**Review by jbankes — APPROVED** [review]

**Comment by mrparkers** [comment]
Before we merge this, it might be a good idea to stand up a new instance of CAF using these changes, just to ensure that we don't hit any name length limits.  I know that we've hit these in the past while attempting to follow some of Azure's naming standards.

Just a suggestion though!

**Comment by anthonygauthier** [comment]
> Before we merge this, it might be a good idea to stand up a new instance of CAF using these changes, just to ensure that we don't hit any name length limits. I know that we've hit these in the past while attempting to follow some of Azure's naming standards.
> 
> Just a suggestion though!

Great idea! It's one of the main reasons why I haven't merged just yet. Trying to test everything as much as possible first so nothing breaks.

**Review by adrielp — APPROVED** [review]

**Review by djtjwillia — APPROVED** [review]
Approving with the understanding we are following Parker's good advice to run an end to end test before we merge

**Comment by ChrisSchreiber** [comment]
I think the resource type prefixes are good but i'm not sure about including "environment" in all of the resource names. Most of the resources are not part of an environment like "dev" or "qa" or "prod" because they are part of a framework used to manage those environments (especially in the foundation modules) Maybe it would make more sense to just hard code those as "caf" or something and then have environment vars for the shared service and landing zones.

**Review by djtjwillia — APPROVED** [review]

**Pull request merged**
